core/rawdb/accessors_metadata.go

```diff
@@ -17,11 +17,67 @@
 package rawdb
 
 import (
+	"encoding/json"
+	"errors"
+
 	"github.com/XinFinOrg/XDPoSChain/common"
 	"github.com/XinFinOrg/XDPoSChain/ethdb"
 	"github.com/XinFinOrg/XDPoSChain/log"
+	"github.com/XinFinOrg/XDPoSChain/params"
+	"github.com/XinFinOrg/XDPoSChain/rlp"
 )
 
+// ReadDatabaseVersion reads the version number from db.
+func ReadDatabaseVersion(db ethdb.KeyValueReader) int {
+	var vsn uint
+	enc, _ := db.Get([]byte("BlockchainVersion"))
```

Comment:
Please change "BlockchainVersion" to "DatabaseVersion" and modify the file `eth/backend.go` as PR #16666.